gemrb.1
.\"Title and section
.TH GEMRB 1

.\"###################################################
.SH NAME 
.B gemrb
\- emulator for Infinity Engine-based games

.\"###################################################
.SH SYNOPSIS
.B gemrb
[-c config-file]
.br
.B torment
.br

.\"###################################################
.SH DESCRIPTION
.I gemrb
is an emulator for Infinity Engine-based games, fine RPGs like Baldur's Gate,
Icewind Dale or Planescape: Torment. 

.I gemrb
is only an engine; to actually play anything you have to have 
data from the original games installed where they can be found by 
.IR gemrb .

.\"###################################################
.SH OPTIONS
.TP
.B \-c
.I FILE
.br
use other configuration file than default
.IR gemrb.cfg .  

.B Note:
You can also use program's name as a mean to select the configuration file.
For example, if the program's name is 
.I torment 
instead of
.IR gemrb ,
the engine first seaches for 
.I torment.cfg 
and only if it's not found it searches for 
.IR gemrb.cfg .

To use this feature, just create a symbolic link
.I torment
pointing to
.I gemrb
binary and run
.IR torment .

.\"###################################################
.SH Configuration File
.PD 0
Configuration file consists of pairs
.IR PARAMETER=VALUE ,
each on its own line. The case of PARAMETER is not significant, 
GameType is the same as GAMETYPE. If the same PARAMETER is defined 
more than once, only the last occurence is used.

Empty lines and lines starting with `#' (hash sign) are ignored.

.B Parameters:

.TP
.BR GameType =(bg1|bg2|iwd|iwd2|how|pst)
Type of the game. The value is a name of subdirectory in override/ 
and GUIScripts/ directories
with game type-specific files and scripts. It is at present also used
for various tweaks hardcoded in GemRB engine.

.TP
.BR GameName =STRING
Arbitrary name of the game. It will be displayed in the window's title bar.

.TP
.BR GamePath =PATH
Path to the directory where the original game is installed. If you can't
install the game under MS Windows or with WINE, you can try to unpack the data files
with the
.I unshield
program found at 
.IR http://synce.sourceforge.net .

.TP
.BR CD1 =PATH
.TP
.BR CD2 =PATH
.TP
.BR CDn =...
Path to the data files for the game's particular CDs.

.\".TP
.\".BR INIConfig =FILE
.\"Name of the game's INI file, relative to 
.\".IR GamePath .
.\"E.g. for Baldur's Gate game it is baldur.ini.


.TP
.BR Width =INT
Game window width (in pixels).

.TP
.BR Height =INT
Game window height (in pixels).

.TP
.BR Bpp =INT
Color depth of the game window (in bits per pixel).

.TP
.BR Fullscreen =(0|1)
Whether the game should run in fullscreen mode. (Note that fullcreen mode is
currently buggy).

.\".TP
.\".BR ForceStereo =(0|1)
.\"Some older games such as BG1 and PST need this parameter to be 
.\".IR 1 .
.\"For other game types set it to
.\".IR 0 .

.TP
.BR TooltipDelay =INT
Delay (in ms) before tooltips are displayed when the mouse is not moving.
Reasonable number for this option is e.g.
.IR 500 .

.TP
.BR VolumeAmbients =INT
Volume of ambient sounds in percents. The default is
.IR 100 .

.TP
.BR VolumeMovie =INT
Volume during movie playback in percents. The default is
.IR 100 .

.TP
.BR VolumeMusic =INT
Volume of background music in percents. The default is
.IR 100 .

.TP
.BR VolumeSFX =INT
Volume of sound effects in percents. The default is
.IR 100 .

.TP
.BR VolumeVoices =INT
Volume of PC or NPC voices in percents. The default is
.IR 100 .


.TP
.BR SkipIntroVideos =(0|1)
This parameter is meant for developers. If set to
.IR 1 ,
the intro and logo videos are skipped to save developer's nerves. The default is
.IR 0 .

.TP
.BR DrawFPS =(0|1)
This parameter is meant for developers. If set to
.IR 1 ,
the current FPS (Frames per Second) value is drawn in the top left window corner. The default is
.IR 0 .

.TP
.BR FogOfWar =(0|1)
This parameter is meant for developers. If set to
.IR 1 ,
the unexplored parts of an area are blacked out. Since the unexplored bitmap is
not updated by actors yet, it's set to
.I 0 
by default.

.TP
.BR EnableCheatKeys =(0|1)
This parameter is meant for developers. If set to
.IR 1 ,
certain keys allow you to inspect the internal state of objects, do forbidden things, etc.
The keys are listed in the
.I CheatKeys.txt
file. Do NOT use this option unless you want to make your hands dirty :-). The default is
.IR 0 .

.TP
.BR ScriptDebugMode =(n)
This parameter is meant for developers. It is a combination of bit values

.IR 1
- count references,

.IR 2
- display cutscene warnings,

.IR 4
- display variable warnings,

.IR 8
- display action warnings,

.IR 16
- display trigger warnings.

The default is
.IR 0 .

.TP
.BR CaseSensitive =(0|1)
When 
this parameter is set to 
.IR 1 ,
the engine will try to find files regardless of their names' case at the cost
of a slight speed penalty. This is especially important when using
game files on CD-ROMs, where the files can't be renamed. Like 
many Windows programs, the original games use inconsistent mix
of lower/upper case letters for naming and referencing their files. 

Set this parameter to 
.I 1
on Unix-like systems.

.TP
.BR SaveAsOriginal =(0|1)
Set this parameter to
.IR 1 ,
if you want to keep the save game compatible with the original engine.

.TP
.BR CachePath =PATH
User-writable directory where the engine caches uncompressed and temporary
files.

.TP
.BR GemRBPath =PATH
Path to the GemRB installation. (To be obsoleted).

.TP
.BR GUIScriptsPath =PATH
Path to the directory with scripts controlling the game's GUI. 

.TP
.BR SavePath =PATH
Path to the directory with save games. Note that this directory has to
contain 
.I save/
and/or eventually
.I mpsave/
subdirectories and that only these subdirectories contain the actual
saved games.

.I SavePath
defaults to
.IR GamePath .

.TP
.BR GameOnCD =(0|1)
defaults to
.IR 0 .


.TP
.BR GameDataPath =PATH
Path to the original game's installed data files, relative to GamePath.
Usually it's 
.IR data .


.TP
.BR GameOverridePath =PATH
Path to the original game's override dir, relative to GamePath.
Usually it's 
.IR override .
Original games use this directory to place last-minute patches.

.TP
.BR GemRBOverridePath =PATH
Path where gemrb looks for its game override directory.
It defaults to
.IR GemRBPath .
The original games hardcoded this information in the engine.


.\"###################################################
.SH FILES
.PD 0
.TP 20
.B /usr/local/etc/gemrb.cfg
system-wide configuration file

.TP
.B /usr/local/share/gemrb/override/
directory with 
.IR gemrb -distributed
data files. These used to be hardcoded in the original engine. The files
are sorted into directories for specific games.

.TP
.B /usr/local/share/gemrb/GUIScripts/
directory with Python scripts providing GUI setup and interaction. The files
are sorted into directories for specific games.

.TP
.B ~/.gemrb/gemrb.cfg
.TP
.B ~/.gemrb/override/
.TP
.B ~/.gemrb/GUIScripts/
user's own configuration and data files.
.TP
.B ~/.gemrb/cache/
cache directory
.PD

.\"###################################################
.SH BUGS
Many.
.I gemrb
is still in an early development stage and not nearly complete. 

If you stumble over something which is supposed to work, either make a bug report
at http://sourceforge.net/projects/gemrb or come to IRC channel 
.I #GemRB
at FreeNode.

.\"###################################################
.SH AUTHOR
The GemRB Project development team at http://gemrb.sourceforge.net

.\"###################################################
.SH COPYING
Copyright (C) 2003-2008 The GemRB Project

This program is free software; you can redistribute it and/or
modify it under the terms of the GNU General Public License
as published by the Free Software Foundation; either version 2
of the License, or (at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.

.\"###################################################
.\"End of file gemrb.man


NEWS
GemRB V0.3.0 (2008-02-17):
  New features:
    - TLK override handling (custom biographies and map notes)
    - weapon immunities
    - party AI
    - expansion playmode
    - more actions, triggers and effects
    - loading of projectile explosion animations
    - kit information window
    - optional CMake build system (windows only)

  Improved features:
    - sound (now perfect!)
    - character generation
    - opcodes
    - character record window
    - pathfinding
    - tooltip delay
    - bugfixes

GemRB V0.2.9 (2007-07-06):
  New features:
    - thieving
    - tracking
    - graphical feedback (color pulse, blur, mirror image, vvc overlays etc)
    - projectiles
    - spell casting
    - item use
    - challenge rating calculation

  Improved features:
    - more opcodes
    - bugfixes
    - shop/inventory gui

GemRB V0.2.8 (2006-12-24):
  New features:
    - equipment is rendered both on paperdoll and avatar
    - weather (snow/rain) is now rendered

  Improved features:
    - action menus
    - game scripting (actions/triggers)

GemRB V0.2.7 (2006-08-30):
  New features:
    - large animations
    - worldmap travel
    - dialogue portraits
    - translucent shadows option
    - personal space of actors
    - combat
    - many new effects
    - overlay animation

  Improved features:
    - Script fixes
    - Action menus
    - TextScreen
    - doors
    - animated overlays
    - new actions

GemRB V0.2.6 (2005-12-06):
  New features:
    - Effects are in a different plugin
    - DoxyGen docs
    - Wallgroup covers
    - Door triggers
    - Action menus (talk/attack)
    - party/protagonist death handled

  Improved features:
    - Textscreen graphic fixed
    - script workflow
    - compilation and running on different systems (MacOSX, PPC Linux)
    - various leaks/instabilities fixed
    - Saving games
    - inventory screens in many games

GemRB V0.2.5 (2005-08-22):
  New features:
    - Save game
    - Effects are now loaded
    - Equipping effects in items
    - Spawn points in areas
    - Textscreen (scrolled text between chapters)

  Improved features:
    - GameScript is now much more reliable: Action override works, triggers fire once and then get cleared
    - fully working Store screen 
    - fixed padding of message window rows (in dialogs)

GemRB V0.2.4 (2005-05-29):
  New features:
    - Store dialogs (Temple, Inn, Container, Tavern, Store)
    - Fog of war with line of sight
    - Doors block path and line of sight
    - Window frames at higher resolutions
    - Animated buttons (PST portraits, Donation window)
    - Store opens when appropriate
    - Containers

  Improved features:
    - Fixed dialogs
    - new GUIScript functions with documentation
    - Fog of war/door/store related gamescript actions
    - fixed object distance and area variable handling in gamescript
    - other new gamescript actions/triggers
    - Implemented PCs fidget animations

  Documentation:
    - Introduction to writing GUIScript scripts

GemRB v0.2.3 (2005-02-13):
  New features:
    - GUI for most of the games, especially interactive Inventory and Spellbook
    - Map and WorldMap
    - Load screen interstitials with progress bar
    - Spell and item cache to speed up object management
    - Added gamescript actions/triggers 
    - Selection of spells during character generation
    - First attempt on effects code
    - First attempt on Fog-Of-War
    - Tooltips
    - Overhead text
    - Ambient sounds
    - Volume control
    - Manual page gemrb(1)
    - Documentation for GemRB Python API and our custom override files

  Improved features:
    - Character generation
    - GUI
    - Build infrastructure on Linux and Un*x systems
    - Progress towards portability to 64 bit and big endian machines
    - Many bugfixes and new bugs as well ;-)
    - Shortened version numbers
    - Simplified user configuration, game specific settings are now
      in gemrb/override dir


gemrb/GemRB.cpp
/* GemRB - Infinity Engine Emulator
 * Copyright (C) 2003 The GemRB Project
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.

 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.

 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 *
 * $Id$
 *
 */

// GemRB.cpp : Defines the entry point for the application.

#define GEM_APP

#include "stdio.h"
#include "includes/win32def.h"
#include "plugins/Core/Interface.h"
#include "plugins/Core/Video.h"
#include "plugins/Core/ResourceMgr.h"
#include "plugins/Core/MapMgr.h"
#include "plugins/Core/ImageMgr.h"
#include "plugins/Core/DialogMgr.h"
#include "plugins/Core/WindowMgr.h"

#ifndef WIN32
#include <ctype.h>
#include <sys/time.h>
#include <dirent.h>
#else
#include <windows.h>
#endif

//this supposed to convince SDL to work on OS/X
#ifdef __APPLE_CC__ // we need startup SDL here
#include "SDL.h"
#endif

int main(int argc, char* argv[])
{
	core = new Interface( argc, argv );
	if (core->Init() == GEM_ERROR) {
		delete( core );
		printf("Press enter to continue...");
		getc(stdin);
		return -1;
	}
	core->Main();
	delete( core );
#ifndef WIN32
	//reinitialize the console colors.
	printf("\033[%sm","0") ;
#endif
	return 0;
}
